File: programmers/js/new/level2/튜플.js
function solution(s) {
    var answer = [];
    let str = s.substring(1, s.length - 1);
    let isOpen = false;
    let number = "";
    const arr = [[]];
    for (let i = 0; i < str.length; ++i) {
        const c = str[i];
        if (c === "{") {
            isOpen = true;
        } else if (c === "}") {
            arr[arr.length - 1].push(parseInt(number));
            number = "";
            isOpen = false;
        } else {
            if (c === ",") {
                if (isOpen) {
                    arr[arr.length - 1].push(parseInt(number));
                    number = "";
                } else {
                    arr.push([]);
                }
            } else {
                number += c;
            }
        }
    }
    const set = new Set();
    arr.sort((a, b) => a.length - b.length).forEach((row) => {
        row.forEach((num) => {
            if (!set.has(num)) {
                set.add(num);
                answer.push(num);
            }
        });
    });
    return answer;
}

console.log(solution("{{2},{2,1},{2,1,3},{2,1,3,4}}"));

File: programmers/python/level2/튜플.py
def solution(s):
    s = s[1:-1]
    is_open = False
    num = 0
    my_set = set()
    set_list = []
    for c in s:
        if c == '{':
            is_open = True
            num = 0
            my_set = set()
        elif c == '}':
            is_open = False
            my_set.add(num)
            set_list.append(my_set)
        elif c == ',':
            if is_open:
                my_set.add(num)
                num = 0
        else:
            num = num* 10 + int(c)
    set_list.sort(key=lambda e: len(e))
    visited = set()
    answer = []
    for s in set_list:
        for num in s:
            if num not in visited:
                visited.add(num)
                answer.append(num)
    return answer

print(solution("{{2,1},{2},{2,1,3},{2,1,3,4}}"))